<launch>
    <!-- Node Settings -->
    <arg name="output" default="screen"/>
    <arg name="imu" default="imu"/>

    <!-- IMU Settings -->
    <arg name="port" default="/dev/ttyUSB0" />

    <!-- Frame ID for messages -->
    <arg name="frame_id" default="$(arg imu)"/>

    <!-- Baudrate of serial comms (see manual for allowed values) -->
    <arg name="baudrate" default="921600"/>

    <!-- Data rate in Hz -->
    <arg name="imu_rate" default="200"/>

    <!-- Sync out settings -->
    <arg name="enable_sync_out" default = "true"/>
    <arg name="sync_out_rate" default = "30"/>
    <arg name="sync_out_pulse_width" default = "500000"/>

    <!-- Ros Topic settings -->
    <arg name="enable_magnetic_field" default="true"/>
    <arg name="enable_pressure" default="true"/>
    <arg name="enable_temperature" default="true"/>

    <node pkg="imu_vn_100" name="$(arg imu)" type="imu_vn_100_cont_node" output="$(arg output)">
        <param name="port" type="string" value="$(arg port)" />
        <param name="baudrate" type="int" value="$(arg baudrate)" />
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="imu_rate" type="int" value="$(arg imu_rate)" />
        <param name="enable_magnetic_field" type="bool" value="$(arg enable_magnetic_field)" />
        <param name="enable_pressure" type="bool" value="$(arg enable_pressure)" />
        <param name="enable_temperature" type="bool" value="$(arg enable_temperature)" />
        <param name="enable_sync_out" type="bool" value="$(arg enable_sync_out)" />
        <param name="sync_out_rate" type="int" value="$(arg sync_out_rate)" />
        <param name="sync_out_pulse_width" type="int" value="$(arg sync_out_rate)" />
    </node>
</launch>
